projects/vdk-plugins/vdk-meta-jobs/src/vdk/plugin/meta_jobs/meta_dag.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import json
+import logging
+import os
+import pprint
+import sys
+import time
+from graphlib import TopologicalSorter
+from typing import Any
+from typing import Dict
+from typing import List
+
+from taurus_datajob_api import ApiException
+from vdk.plugin.meta_jobs.cached_data_job_executor import TrackingDataJobExecutor
+from vdk.plugin.meta_jobs.meta import TrackableJob
+from vdk.plugin.meta_jobs.remote_data_job_executor import RemoteDataJobExecutor
+from vdk.plugin.meta_jobs.time_based_queue import TimeBasedQueue
+
+log = logging.getLogger(__name__)
+
+
+class MetaJobsDag:
+    def __init__(self, team_name: str):
+        self._team_name = team_name
+        self._topological_sorter = TopologicalSorter()
+        self._delayed_starting_jobs = TimeBasedQueue(
+            min_ready_time_seconds=int(
+                os.environ.get("VDK_META_JOBS_DELAYED_JOBS_MIN_DELAY_SECONDS", 30)
+            ),
+            randomize_delay_seconds=int(
+                os.environ.get(
+                    "VDK_META_JOBS_DELAYED_JOBS_RANDOMIZED_ADDED_DELAY_SECONDS", 600
```

Comment:
randomized seems rather peculiar strategy to use during orchestration

Reply:
It's meant to be randomized backoff delays so that the delay is more randomly spread in time and the API Server is not hammered with requests simultaneously. 

It's hardly a novel thing - the spring re-try framework has similar options - https://docs.spring.io/spring-retry/docs/api/current/org/springframework/retry/annotation/Backoff.html#random--